flagging_site/blueprints/flagging.py

```diff
@@ -72,13 +72,57 @@ def index() -> str:
     returns: render model on index.html
     """
     df = get_data()
-    flags = {
-        2: reach_2_model(df, rows=1)['safe'].iloc[0],
-        3: reach_3_model(df, rows=1)['safe'].iloc[0],
-        4: reach_4_model(df, rows=1)['safe'].iloc[0],
-        5: reach_5_model(df, rows=1)['safe'].iloc[0]
+    
+    # Important: Restructured these dicts for easier use.
+    # Warning - The following aren't implemented yet due to inconsistencies with http://notification.crwa.org/wqmodel/charlesflags.php (old) vs https://www.crwa.org/flagging-program.html (new):
+        # not on old CRWA page, hence not implemented below:
+        # 'Northeastern's Henderson Boathouse'
+        # 'Paddle Boston at Herter Park' -- is this the same as Reach 3's 'CRCK at Herter Park?'
+
+        # only found on old CRWA page only, hence implemented according to old website's format:
+        # 'CRCK at Herter Park'
+        # 'CRCK at Kendall Square'
+
+    homepage = {
+        2: {
+            'flag': reach_2_model(df, rows=1)['safe'].iloc[0],
+            'time': reach_2_model(df, rows=25)['time'].iloc[0],
+            'names': {
+                0: 'Newton Yacht Club',
+                1: 'Watertown Yacht Club',
+                2: 'Community Rowing, Inc.'
+            }
+        },
+        3: {
+            'flag': reach_3_model(df, rows=1)['safe'].iloc[0],
+            'time': reach_3_model(df, rows=25)['time'].iloc[0],
+            'names': {
+                0: 'CRCK at Herter Park', # Found on old CRWA page only
+                1: 'Harvard\'s Weld Boathouse' 
+            }
+        },
+        4: {
+            'flag': reach_4_model(df, rows=1)['safe'].iloc[0],
+            'time': reach_4_model(df, rows=25)['time'].iloc[0],
+            'names': {
+                0: 'Riverside Boat Club'
+            }
+        },
+        5: {
+            'flag': reach_5_model(df, rows=1)['safe'].iloc[0],
+            'time': reach_5_model(df, rows=25)['time'].iloc[0],
```

Comment:
Remove the `time` key from each dict and somewhere in this function block, simply do this:

```python
model_last_updated_time = reach_5_model(df, rows=1)['time'].iloc[0]
```

As a note, The way you're getting the time currently doesn't work. The reason why is subtle, but it's a combination of how `rows=?` works, how the `reach_#_model` functions work, and how Pandas's `iloc` works.

So basically, you're defining `rows=25`. This means we're returning 25 rows of the model. The rows of the model are in ascending order by time. This means the _oldest_ time is at the top of the dataframe, and the _newest_ time is at the bottom of the dataframe.

`iloc[0]` returns the first row of the data. Do you see the problem yet? If you have 25 rows and the oldest data is at the top, you're returning the timestamp from 25 hours ago!

`rows=1` solves this problem because when `rows=1`, then the _only_ row is also the first row, and it also happens to be the _latest_ row too.